internal/cli/action/build.go

```diff
@@ -65,17 +66,19 @@ func Build(ctx *cli.Context) error {
 
 	logger.Info("Validated image configuration")
 
-	outDir, err := createOutputDirectory(system.FS(), args.BuildDir)
+	buildOutput := fmt.Sprintf("build-%s", time.Now().UTC().Format("2006-01-02T15-04-05"))
+	outDir, err := config.CreateOutputDir(system.FS(), args.BuildDir, buildOutput, 0700)
 	if err != nil {
 		logger.Error("Creating build directory failed")
 		return err
 	}
 
 	defer func() {
 		logger.Debug("Cleaning up build-dir %s", outDir)
-		err = system.FS().RemoveAll(string(outDir))
-		if err != nil {
+		rmErr := system.FS().RemoveAll(string(outDir))
+		if rmErr != nil {
 			logger.Error("Cleaning up build-dir %s", outDir)
+			err = errors.Join(err, rmErr)
```

Comment:
This is probably an overkill. Even if this piece fails, the overall "build" process should be successful if `err` is nil here.